.github/workflows/gradle-run-tests.yml

```diff
@@ -7,27 +7,58 @@ on:
 jobs:
   gradle-run-tests:
     runs-on: macos-latest
+    strategy:
+      matrix:
+        api-level: [29]
+
     steps:
       - name: Checkout
         uses: actions/checkout@v2
         with:
           submodules: recursive # Needed in order to fetch Kalium sources for building
           fetch-depth: 0
+
       - name: Set up JDK 11
         uses: actions/setup-java@v2
         with:
           java-version: '11'
           distribution: 'adopt'
           cache: gradle
+
       - name: Validate Gradle wrapper
         uses: gradle/wrapper-validation-action@e6e38bacfdf1a337459f332974bb2327a31aaf4b
+      
+      - name: Gradle cache
+        uses: gradle/gradle-build-action@v2
+        
+      - name: AVD cache
+        uses: actions/cache@v2
+        id: avd-cache
+        with:
+          path: |
+            ~/.android/avd/*
+            ~/.android/adb*
+          key: avd-${{ matrix.api-level }}
+
+      - name: create AVD and generate snapshot for caching
+        if: steps.avd-cache.outputs.cache-hit != 'true'
+        uses: reactivecircus/android-emulator-runner@v2
+        with:
+          api-level: ${{ matrix.api-level }}
+          force-avd-creation: false
+          emulator-options: -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
```

Comment:
does `-camera-back none` mean we won't be able to use the emulator camera to test the different avatar update flows?

Reply:
IDK, we shall find out